core/src/main/java/org/elasticsearch/action/admin/cluster/node/tasks/list/TaskInfo.java

```diff
@@ -0,0 +1,159 @@
+import org.elasticsearch.tasks.Task;
+
+import java.io.IOException;
+
+/**
+ * Information about currently running task
+ */
+public class TaskInfo implements Streamable, ToXContent {
+
+    private String node;
+
+    private long id;
+
+    private String type;
+
+    private String action;
+
+    private String description;
+
+    private String parentNode;
+
+    private long parentId;
+
+    private boolean canceled;
+
+    TaskInfo() {
+
+    }
+
+    public TaskInfo(String node, Task task, boolean detailed) {
+        this.node = node;
+        this.id = task.id();
+        this.type = task.type();
+        this.action = task.action();
+        this.description = detailed ? task.description() : null;
+        this.parentId = task.parentId();
+        this.parentNode = task.parentNode();
+        this.canceled = task.canceled();
+    }
+
+    public TaskInfo(StreamInput in) throws IOException {
+        readFrom(in);
+    }
+
+    public String node() {
+        return node;
+    }
+
+    public long id() {
+        return id;
+    }
+
+    public String type() {
+        return type;
+    }
+
+    public String action() {
+        return action;
+    }
+
+    public String description() {
+        return description;
+    }
+
+    public String parentNode() {
+        return parentNode;
+    }
+
+    public long parentId() {
+        return parentId;
+    }
+
+    public boolean canceled() {
+        return canceled;
+    }
+
+    @Override
+    public void readFrom(StreamInput in) throws IOException {
+        node = in.readString();
+        id = in.readLong();
+        type = in.readString();
+        action = in.readString();
+        description = in.readOptionalString();
+        parentNode = in.readOptionalString();
+        parentId = in.readLong();
+        canceled = in.readBoolean();
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        out.writeString(node);
+        out.writeLong(id);
+        out.writeString(type);
+        out.writeString(action);
+        out.writeOptionalString(description);
+        out.writeOptionalString(parentNode);
+        out.writeLong(parentId);
+        out.writeBoolean(canceled);
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.startObject();
+        builder.field(Fields.NODE, node);
+        builder.field(Fields.ID, id);
+        builder.field(Fields.TYPE, type);
+        builder.field(Fields.ACTION, action);
+        if (description != null) {
+            builder.field(Fields.DESCRIPTION, description);
+        }
+        if (parentNode != null) {
+            builder.field(Fields.PARENT_NODE, parentNode);
+            builder.field(Fields.PARENT_ID, parentId);
+        }
+        builder.endObject();
+        return builder;
+    }
+
+    static final class Fields {
+        static final XContentBuilderString NODE = new XContentBuilderString("node");
```

Comment:
same here these strings are only used in one place just use them directly and trash the Fields class